users/service.go

```diff
@@ -146,13 +139,12 @@ func (svc usersService) UserInfo(ctx context.Context, token string) (User, error
 		Password: "",
 		Metadata: dbUser.Metadata,
 	}, nil
-
 }
 
 func (svc usersService) UpdateUser(ctx context.Context, token string, u User) errors.Error {
-	email, err := svc.idp.Identity(token)
-	if err != nil {
-		return ErrUnauthorizedAccess
+	email, err := svc.identify(ctx, token)
+	if err != nil || email != u.Email {
```

Comment:
This test is not necessary and probably will add a bug. I recently fixed this in the master: https://github.com/mainflux/mainflux/pull/959/files